client/types/util.d.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * @file Misc. type declarations utilized across the repo.
+ */
```

Comment:
They can be imported from a `.d.ts` too, it's just that this is doing `export declare global` because some types get used so often. I was wondering what you thought about that, personally I don't mind the globals.

Reply:
For me there's two situations:

1. Types should be kept in the sub-repository that they pertain to, except where they really do need to be cross-repository in which case they should be kept a level up. This results in lots of type definition files across several subrepos, so it's hard to tell at a glance which global definitions file a given type is coming from. In that case it's easier to go for standard types files with explicit imports.
2. There is a single global types files for all the subrepos (this is the current situation). In this case it's hard to know from the types file which if any are being used, and if so, which subrepos are using them.

Either way, for me it boils down to explicit imports being better in the long run.

Right now it's another of those things that is basically fine for now up until it isn't. It's not a big deal to navigate or to fix while there are so few of them, but if this file becomes much bigger I'd like to see it become more explicit.

Reply:
An alias could work here. Aliasing this folder to `@types` would let you get easy imports to all the common types. Does that sound good? (if so let's do that in another commit/PR, this is already getting kinda messy for me on my machine)

Also, FYI: I didn't want to make this a package because it wouldn't actually like, compile to anything? It would just be filler in the module list.

Reply:
I've said before that I think we should have our aliases live under `@wikijump` for standard naming, as there will definitely be further shared libraries in the future. So this would be `@wikijump/types`.